src/App.js

```diff
@@ -2,19 +2,36 @@ import * as React from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
+// Custom hook for semi-persistent state
+function useSemiPersistentState(key, initialState) {
+  const [state, setState] = React.useState(
+    localStorage.getItem(key) || initialState
+  );
+
+  React.useEffect(() => {
+    localStorage.setItem(key, state);
+  }, [key, state]);
```

Comment:
You should stringify the value here when saving it to localStorage, otherwise every time you need / want to display in your application you will need to parse it.
```
React.useEffect(() => {
  localStorage.setItem(key, JSON.stringify(state))
}, [key, state])
```